// @storybook/addon-links/dist/preview.test.js
"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.mockChannel = void 0;

require("core-js/modules/es6.promise");

require("regenerator-runtime/runtime");

var _addons = _interopRequireDefault(require("@storybook/addons"));

var _preview = require("./preview");

var _ = require(".");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

jest.mock('@storybook/addons');

var mockChannel = function mockChannel() {
  var cb;
  return {
    emit: function emit(id, payload) {
      if (id === _.REQUEST_HREF_EVENT_ID) {
        cb("?selectedKind=".concat(payload.kind, "&selectedStory=").concat(payload.story));
      }
    },
    on: function on(id, callback) {
      if (id === _.RECEIVE_HREF_EVENT_ID) {
        cb = callback;
      }
    }
  };
};

exports.mockChannel = mockChannel;
describe('preview', function () {
  describe('linkTo()', function () {
    it('should select the kind and story provided', function () {
      var channel = {
        emit: jest.fn()
      };

      _addons.default.getChannel.mockReturnValue(channel);

      var handler = (0, _preview.linkTo)('kind', 'story');
      handler();
      expect(channel.emit).toHaveBeenCalledWith(_.EVENT_ID, {
        kind: 'kind',
        story: 'story'
      });
    });
    it('should handle functions returning strings', function () {
      var channel = {
        emit: jest.fn()
      };

      _addons.default.getChannel.mockReturnValue(channel);

      var handler = (0, _preview.linkTo)(function (a, b) {
        return a + b;
      }, function (a, b) {
        return b + a;
      });
      handler('foo', 'bar');
      expect(channel.emit).toHaveBeenCalledWith(_.EVENT_ID, {
        kind: 'foobar',
        story: 'barfoo'
      });
    });
  });
  describe('hrefTo()', function () {
    it('should return promise resolved with story href',
    /*#__PURE__*/
    _asyncToGenerator(
    /*#__PURE__*/
    regeneratorRuntime.mark(function _callee() {
      var channel, href;
      return regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              channel = mockChannel();

              _addons.default.getChannel.mockReturnValue(channel);

              _context.next = 4;
              return (0, _preview.hrefTo)('kind', 'story');

            case 4:
              href = _context.sent;
              expect(href).toBe('?selectedKind=kind&selectedStory=story');

            case 6:
            case "end":
              return _context.stop();
          }
        }
      }, _callee, this);
    })));
  });
});
