// @rowno/sparkline/commonjs/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _d3Shape = require('d3-shape');

var _d3Scale = require('d3-scale');

var _d3Array = require('d3-array');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var curveFunction = _d3Shape.curveMonotoneX;
var defaultColors = {
  area: 'rgba(199, 228, 255, 0.5)',
  line: '#004585'
};

var Sparkline = function (_PureComponent) {
  (0, _inherits3.default)(Sparkline, _PureComponent);

  function Sparkline() {
    (0, _classCallCheck3.default)(this, Sparkline);
    return (0, _possibleConstructorReturn3.default)(this, (Sparkline.__proto__ || (0, _getPrototypeOf2.default)(Sparkline)).apply(this, arguments));
  }

  (0, _createClass3.default)(Sparkline, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          lines = _props.lines,
          width = _props.width,
          height = _props.height;

      var maxX = (0, _d3Array.max)(lines, function (s) {
        return s.values.length - 1;
      });
      var maxY = (0, _d3Array.max)(lines, function (s) {
        return (0, _d3Array.max)(s.values);
      });
      var x = (0, _d3Scale.scaleLinear)().domain([0, maxX]).range([0, width]);
      // Set range to 1 to make room for the line stroke
      var y = (0, _d3Scale.scaleLinear)().domain([0, maxY]).range([height, 1]);

      var areaFunction = (0, _d3Shape.area)().x(function (d, i) {
        return x(i);
      }).y0(height).y1(function (d) {
        return y(d);
      }).curve(curveFunction);

      var lineFunction = (0, _d3Shape.line)().x(function (d, i) {
        return x(i);
      }).y(function (d) {
        return y(d);
      }).curve(curveFunction);

      return _react2.default.createElement(
        'svg',
        { width: width, height: height, viewBox: '0 0 ' + width + ' ' + height },
        lines.map(function (line, index) {
          var colors = (0, _extends3.default)({}, defaultColors, line.colors);
          return _react2.default.createElement(
            'g',
            { key: line.key || index },
            _react2.default.createElement(
              'path',
              { d: areaFunction(line.values), fill: colors.area },
              line.title && _react2.default.createElement(
                'title',
                null,
                line.title
              )
            ),
            _react2.default.createElement('path', { d: lineFunction(line.values), stroke: colors.line, fill: 'none' })
          );
        })
      );
    }
  }]);
  return Sparkline;
}(_react.PureComponent);

Sparkline.displayName = 'Sparkline';
Sparkline.propTypes = {
  lines: _propTypes2.default.arrayOf(_propTypes2.default.shape({
    values: _propTypes2.default.arrayOf(_propTypes2.default.number.isRequired).isRequired,
    colors: _propTypes2.default.shape({
      area: _propTypes2.default.string.isRequired,
      line: _propTypes2.default.string.isRequired
    }),
    title: _propTypes2.default.string,
    key: _propTypes2.default.any
  })).isRequired,
  width: _propTypes2.default.number.isRequired,
  height: _propTypes2.default.number.isRequired
};
exports.default = Sparkline;
